### ISSUE TITLE
Use hilbert curves to improve locality and speed for LocationIndex

### ISSUE THREAD

**Issue by karussell** [graphhopper/graphhopper#1840]
We currently use a Quadtree for the LocationIndex and [binary Geohashes / spatial keys](https://karussell.wordpress.com/2012/05/23/spatial-keys-memory-efficient-geohashes/). Instead of binary Geohashes we could use [hilbert curves](http://blog.notdot.net/2009/11/Damn-Cool-Algorithms-Spatial-indexing-with-Quadtrees-and-Hilbert-Curves) to improve locality and likely also speed for many points.

**Comment by otbutz** [comment]
Did you test Quadtree vs R-Tree?

**Comment by karussell** [comment]
No, never tested this. But as we already have a highly optimized data structure (in terms of memory usage, construction time and query speed even for the world wide case) it will be hard to beat this and it is currently only significant for many query points.
